source/slang/slang-emit-hlsl.cpp

```diff
@@ -1274,6 +1273,13 @@ void HLSLSourceEmitter::emitVarDecorationsImpl(IRInst* varDecl)
             m_writer->emit(")]]\n");
             continue;
         }
+        if (auto collection = as<IRMemoryQualifierSetDecoration>(decoration))
```

Comment:
Feel free to ignore this comment.
But I will try to stay away from the term "collection", because the meaning is too vague.
I like to suggest `qualifierSet` for this case.